src/include/miopen/par_for.hpp

```diff
@@ -91,9 +91,10 @@ void par_for_impl(std::size_t n, std::size_t threadsize, F f)
         const std::size_t grainsize = std::ceil(static_cast<double>(n) / threads.size());
 
         std::size_t work = 0;
-        std::generate(threads.begin(),
-                      threads.end(),
-                      std::bind(thread_factory{}, std::ref(work), n, grainsize, f));
+        std::generate(threads.begin(), threads.end(), [&]() {
+            return thread_factory()(work, n, grainsize, f);
+        });
+        // cppcheck-suppress unsignedLessThanZero
```

Comment:
Why is this needed? It doesn't currently fail with cppcheck. Are you using a different version of cppcheck?

Reply:
Ah wait, I think I might see the issue. Because you switch to using a lambda, cppcheck doesn't recognize how the `work` and `n` variable is changed.

Reply:
Exactly. I'll better disable tidy check globally. Actually I like this too. Exchanging one tidy warning to two cppcheck ones only to disable two instead of one looks weird))